form-link/src/main/java/com/ritense/formlink/service/impl/CamundaFormAssociationService.java

```diff
@@ -453,4 +458,17 @@ private ObjectNode buildMetaDataObject(Document document) {
         metaDataNode.put("sequence", document.sequence());
         return metaDataNode;
     }
+
+    private static <T> Collector<T, ?, Optional<T>> single() {
```

Comment:
Not entirely happy with this name, as it doesn't fully convey what this method is supposed to return. It returns a collector that results in one result if the list only has one entry. That's not clear when reading the method name.

Reply:
Do you have a suggestion for improving the name? from the stackoverflow I see a couple:

- onlyElement see [
](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/MoreCollectors.html#onlyElement--)- singleElementCollector
- zeroOrOne
- onlyOne
- toSingleton

Reply:
singleElementCollector sounds best out of these options. I was playing with names like singleElementIfOnlyOneElementPresentCollector, but that becomes too cumbersome in my opinion.

Reply:
singleElementCollector is fine then.